doc/src/guides/solving/solve-numerically.md

```diff
@@ -0,0 +1,213 @@
+
+As noted above, SymPy focuses on symbolic computation and is not optimized for
+numerical calculations. A recommend use case for SymPy in a workflow is as given
+below:
+- use SymPy to generate (by symbolically simplifying or solving an equation) the
+  mathematical expression
+- use a numerical library such as SciPy to generate numerical solutions
+
+```py
+>>> from sympy import simplify, cos, sin
+>>> from sympy.abc import x, y
+>>> expr = cos(x * (x + x**2)/(x*sin(y)**2 + x*cos(y)**2 + x))
+>>> simplify(expr)
+cos(x*(x + 1)/2)
+>>> from sympy.utilities.lambdify import implemented_function
+>>> from sympy import lambdify
+>>> f = implemented_function('f', lambda x: cos(x*(x + 1)/2))
+>>> lam_f = lambdify(x, f(x))
+>>> from scipy import optimize
+>>> sol = optimize.root_scalar(lam_f, bracket=[0, 2], method='brentq')
+>>> sol.root
+1.3416277185114782
+```
+
+## Use the Solution Result
+
+### Substitute the Result Into an Expression
+
+The best practice is to use {func}`~sympy.core.evalf` to to substitute numerical
+values into expressions. Using [`subs`](sympy.core.basic.Basic.subs) can give
+incorrect result due to precision errors:
+
+```py
+>>> from sympy import cos, nsolve, Symbol
+>>> x = Symbol('x')
+>>> f = cos(x) - x
+>>> x_value = nsolve(f, x, 1); x_value
+0.739085133215161
+>>> y = Symbol('y')
+>>> z = Symbol('z')
+>>> values = {x: x_value, y: 1e16, z: 1e16}
+>>> (x + y - z).evalf(subs=values) # Gives correct result
+0.739085133215161
+>>> (x + y - z).subs(values) # Gives incorrect result due to precision errors
+0
+```
+
+You can also leave some symbols as variables:
+
+```py
+>>> from sympy import cos, nsolve, Symbol
+>>> x = Symbol('x')
+>>> f = cos(x) - x
+>>> x_value = nsolve(f, x, 1); x_value
+0.739085133215161
+>>> y = Symbol('y')
+>>> z = Symbol('z')
+>>> g = x * y**2
+>>> values = {x: x_value, y: 1}
+>>> (x + y - z).evalf(subs=values)
+1.73908513321516 - z
+```
+
+## Not all Equations Can be Solved
+
+{func}`~.nsolve` is a numerical solving function, so it can often provide a
+solution for equations which cannot be solved algebraically.
+
+### Equations With no Solution
+
+Some equations have no solution, in which case SymPy may return an error. For
+example, the equation $e^x = 0$ (`exp(x)` in SymPy) has no solution:
+
+```py
+>>> from sympy import nsolve, exp
+>>> from sympy.abc import x
+>>> nsolve(exp(x), x, 1, prec=20)
+Traceback (most recent call last):
+...
+ValueError: Could not find root within given tolerance. (5.4877893607115270300540019e-18 > 1.6543612251060553497428174e-24)
+Try another starting point or tweak arguments.
+```
+
+## Report a Problem
+
+If you find a problem with {func}`~.nsolve`, please post it on the [mailing
+list](https://groups.google.com/g/sympy), or open an issue on [SymPy's GitHub
+page](https://github.com/sympy/sympy/issues). Until the issue is resolved, you
+can use NumPy's {external:func}`numpy.linalg.solve` or SciPy's
+{external:func}`scipy.linalg.solve`.
```

Comment:
The proper analogue of `nsolve` is SciPy's `fsolve` because `nsolve` is not specifically for linear equations.